src/cryptsetup/cryptsetup-generator.c

```diff
@@ -541,7 +541,7 @@ static int parse_proc_cmdline_item(const char *key, const char *value, void *dat
                 if (proc_cmdline_value_missing(key, value))
                         return 0;
 
-                d = get_crypto_device(startswith(value, "luks-") ? value+5 : value);
+                d = get_crypto_device(startswith(value, "luks-") ? value+strlen("luks-") : value);
```

Comment:
actually startswith() returns the pointer after the prefix. Hence, by using gcc's syntax extension that the 2nd/2rd ternary op can be ommitted, we can make this even perttier:

```c
d = get_crypto_device(startswith(value, "luks-") ?: value);
```

could you rework the patch like that pls?

Reply:
Thanks, updated!